subsys/bluetooth/audio/has.c

```diff
@@ -77,45 +457,91 @@ static ssize_t write_control_point(struct bt_conn *conn, const struct bt_gatt_at
 	return len;
 }
 
-static ssize_t read_active_preset_index(struct bt_conn *conn, const struct bt_gatt_attr *attr,
-					void *buf, uint16_t len, uint16_t offset)
+int bt_has_preset_register(struct bt_has_preset_register_param *param)
 {
-	BT_DBG("conn %p attr %p offset %d", (void *)conn, attr, offset);
+	struct bt_has_preset *preset = NULL;
 
-	if (offset > sizeof(has.active_index)) {
-		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
+	CHECKIF(param == NULL) {
+		BT_ERR("param is NULL");
+		return -EINVAL;
 	}
 
-	return bt_gatt_attr_read(conn, attr, buf, len, offset, &has.active_index,
-				 sizeof(has.active_index));
+	CHECKIF(param->index == BT_HAS_PRESET_INDEX_NONE) {
+		BT_ERR("param->index is invalid");
+		return -EINVAL;
+	}
+
+	CHECKIF(param->name == NULL) {
+		BT_ERR("param->name is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(strlen(param->name) < BT_HAS_PRESET_NAME_MIN) {
```

Comment:
Consider caching `strlen(param->name)` to avoid calling the function multiple times (which usually does a loop until it reaches a NULL terminator).